src/extensions/default/PrefsCodeHints/main.js

```diff
@@ -38,11 +38,24 @@ define(function (require, exports, module) {
         JSONUtils           = brackets.getModule("language/JSONUtils"),
         Strings             = brackets.getModule("strings"),
         ThemeManager        = brackets.getModule("view/ThemeManager"),
+        CodeInspection      = brackets.getModule("language/CodeInspection"),
         _                   = brackets.getModule("thirdparty/lodash"),
-        data                = JSON.parse(require("text!data.json")),
+        languages           = LanguageManager.getLanguages(),
```

Comment:
@sprintr I still don't see the point of having a cache of something that could be grabbed at the point of getting the hints like it is done with the code hint providers. `getLanguages` copies the object, but since you will have to iterate through the list eventually to create the ui, should be fine to get them on `getHints` when you actually need them. And the same goes for the themes, and maybe even the preferences. Seems easier than keeping a copy that needs to be synched.